### ISSUE TITLE
SereneTweaks support for 1.7.10

### ISSUE THREAD

**Issue by DarkShadow44** [DarkShadow44/SereneSeasons#27]
See https://github.com/FIREdog5/SereneTweaks/issues/8

**Comment by DarkShadow44** [comment]
To build just "gradlew jar" should be enough. I already have an untested port pushed, maybe you can test that.

**Comment by danielrossyamitrano** [comment]
I'm getting an error while trying to build from source at it says at the readme.

java.lang.IllegalArgumentException: Could not determine java version from '16.0.2'.
        at org.gradle.api.JavaVersion.toVersion(JavaVersion.java:63)
        at org.gradle.api.JavaVersion.current(JavaVersion.java:72)
        at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:32)
        at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:206)
        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
        at org.gradle.launcher.Main.doAction(Main.java:33)
        at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:54)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:35)
        at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:30)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:127)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:58)

From what I gather I have to update my version of gradle, but I don't really know.

**Comment by danielrossyamitrano** [comment]
Nevermind. I didn't have a JAVA_HOME path created.

**Comment by danielrossyamitrano** [comment]
AHHHHHHHHHHHH I still have problems. It was working fine until it relized that I didn't have a Java compiler so I installed JDK 18.0.1.1 (the lastest I could find) and now the `gradlew reobf` refuses to work.

**Comment by danielrossyamitrano** [comment]
Finally, I got the gradle build working. The thing I needed was a x86 compiler (I was using a x64, that apparently doesn't work with Java 8). I will test it tomorrow because now it is 2:11 AM and I'm just tired.

**Comment by danielrossyamitrano** [comment]
So far, so good. I started a new Large Biomes world sucessfully and after an hour of play I haven't had any problems (I haven't crafted the clock yet) related to the mod. I have one question tho: this backdrop doesn't come with Serene Tweeks installed, isn't it?

**Comment by DarkShadow44** [comment]
It should come with SereneTweaks, though I didn't test whether it actually works ;)

**Comment by danielrossyamitrano** [comment]
Oh, I see. That is what you want me to test, right?

**Comment by danielrossyamitrano** [comment]
![2022-07-16_00 34 34](https://user-images.githubusercontent.com/8364907/179339707-5feab7a2-88c9-4374-a0f7-e4f4b51bb46a.png)
I have almost every kind of tree planted around my house (except acacia) and I found that the Spruce Trees' foliage doesn't seem to be changing its colors whereas the Birch trees (pictured) as well as the Rubber Trees (not pictured) from Industrial Craft 2 experimental do change.

I don't know if that's a bug or a feature since Spruce trees grow normally in the extreme hills biome which is cold by default. And I think the real trees of that variant don't loose their leaves in autum. Is that correct?

**Comment by DarkShadow44** [comment]
Yes, Spruce is unaffected by seasons, always has been.

**Comment by danielrossyamitrano** [comment]
Oh, I see. That's what I thought.

**Comment by danielrossyamitrano** [comment]
A few shots from the Plains biome my home is, at nightime. I walked a few chunks away and the snowing cotinued.
https://ibb.co/4d3JhYm
https://ibb.co/Mk0ss35
https://ibb.co/qsV0JTm
https://ibb.co/0D3ypZ6
https://ibb.co/74dtX6x
https://ibb.co/9WH70M6

And finally four shots at daytime from high above a platform.
https://ibb.co/r7LzR66
https://ibb.co/smVKtk9
https://ibb.co/kxkMPmB
https://ibb.co/KmkrD5H

**Comment by DarkShadow44** [comment]
Of course it continues snowing, it's still winter. SereneTweaks would make it so that there's already snow on the floor when you visit a chunk in winter where you haven't been for a while.

**Comment by danielrossyamitrano** [comment]
I about 7000 meters away from my home, when I think winter ended (I didn't have the clock with me) and all the chunks around me started glitching out of existence. I tried to fix the problem by tweeking the render distance, turning off fancy graphics and so on, but nothing happened. The problem only fixed itself when I removed SereneSeasons from the mods folder, but the snow left by winter is still here.

**Comment by danielrossyamitrano** [comment]
Well, no. It seems to be a problem with other mod since I restarted a world without sereneseasons and the glitch I mentioned still happens.

**Comment by DarianLStephens** [comment]
Any chance you might be able to privately share a compiled version with me on Discord or something? I could test this in a minute.

**Comment by DarkShadow44** [comment]
Sure, just send your discord tag to my email (visible in my github profile)

**Comment by DarianLStephens** [comment]
Having run a version of it for a bit on a server, I haven't seen this work.
The closest I _think_ I've seen is single blocks of ice in water. Maybe it's only working on the first block in a chunk?

**Comment by DarkShadow44** [comment]
That shouldn't happen, although I never tested it in detail.

**Comment by DarianLStephens** [comment]
It's hard to find good screenshots of it, but you can see a single snow block in the center of all these chunks in JourneyMap:
![image](https://user-images.githubusercontent.com/15375876/203508739-50dcd2fb-4ea4-4ad6-9dbf-ae564f15d836.png)

**Comment by DarkShadow44** [comment]
Sorry for the super late solution, but it should be fixed in Fixed in 1.3.0. Please retest, and report back if there's issues.

**Comment by DarianLStephens** [comment]
I'm not 100% sure it's SereneSeasons, but I keep seeing patterns like this in the SereneTweaks-generated snow, where it causes plant generation to fail somehow, being cut off at chunk borders.
![2023-05-15_05 58 13](https://github.com/DarkShadow44/SereneSeasons/assets/15375876/e408e085-0ad8-4b36-b83e-04e096eaf9fa)
It only seems to happen in freshly-generated chunks. Maybe the snow is generating too fast or something?
Here's an example in another world:
![2023-05-14_06 15 00](https://github.com/DarkShadow44/SereneSeasons/assets/15375876/f70e944c-b55e-4230-9f83-42a7a2f14a12)
![2023-05-14_06 15 27](https://github.com/DarkShadow44/SereneSeasons/assets/15375876/72559c91-c35b-48e0-a300-b80f72602c1b)

**Comment by DarkShadow44** [comment]
I'm not exactly sure what the problem is... Can you please open a new issue for that, and show the problem from normal first person view?

**Comment by DarkShadow44** [comment]
I did a few changes, now it uses world time over real system time, and chunks really need to be unloaded for the effect to work. Before it sometimes didn't properly update the time, so that could have had that result...
Could you please retest?

**Comment by danielrossyamitrano** [comment]
> I did a few changes, now it uses world time over real system time, and chunks really need to be unloaded for the effect to work. Before it sometimes didn't properly update the time, so that could have had that result... Could you please retest?

I will do it shortly. Thanks.

**Comment by danielrossyamitrano** [comment]
Well, I don't know why this is closed, but I comment anyway. I don't know if it is a bug or a feature, but the duration of the day/night cycle seems to be the same for the standard minecraft experience. When I commented that I would be nice to have the SereneTweek features in 1.7.10, I was refferering precisely about that.
On a later try, I changed the "Day Duration" setting on "time settings" to 1728000 ticks (which is 24 real time hours) and when I tested it, I noticed that it doesn't work as SereneTweeks does. If I start a new game now (18:10) it should be 18:10 in the game, but it's not, and the sun still sets and rises normallly.
